cmd/app/settings_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022-2025 Salesforce, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	"testing"
+
+	"github.com/slackapi/slack-cli/internal/app"
+	"github.com/slackapi/slack-cli/internal/config"
+	"github.com/slackapi/slack-cli/internal/prompts"
+	"github.com/slackapi/slack-cli/internal/shared"
+	"github.com/slackapi/slack-cli/internal/shared/types"
+	"github.com/slackapi/slack-cli/internal/slackerror"
+	"github.com/slackapi/slack-cli/internal/slacktrace"
+	"github.com/slackapi/slack-cli/test/testutil"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/mock"
+)
+
+func Test_App_SettingsCommand(t *testing.T) {
+	testutil.TableTestCommand(t, testutil.CommandTests{
+		"requires a valid project directory": {
+			ExpectedError: slackerror.New(slackerror.ErrInvalidAppDirectory),
+		},
+		"errors for rosi applications": {
+			Setup: func(t *testing.T, ctx context.Context, cm *shared.ClientsMock, cf *shared.ClientFactory) {
+				cf.SDKConfig.WorkingDirectory = "."
+				projectConfigMock := config.NewProjectConfigMock()
+				projectConfigMock.On(
+					"GetManifestSource",
+					mock.Anything,
+				).Return(
+					config.ManifestSourceLocal,
+					nil,
+				)
+				cm.Config.ProjectConfig = projectConfigMock
+				manifestMock := &app.ManifestMockObject{}
+				manifestMock.On(
+					"GetManifestLocal",
+					mock.Anything,
+					mock.Anything,
+					mock.Anything,
+				).Return(
+					types.SlackYaml{
+						AppManifest: types.AppManifest{
+							Settings: &types.AppSettings{FunctionRuntime: types.SlackHosted},
+						},
+					},
+					nil,
+				)
+				cm.AppClient.Manifest = manifestMock
+			},
+			ExpectedError: slackerror.New(slackerror.ErrAppHosted),
+		},
+		"requires an existing application": {
```

Comment:
note: Nice, I tested this manually and I'm happy to see you already added a test!